src/value_classes/value_id.rs

```diff
@@ -405,7 +468,11 @@ impl cmp::Eq for ValueID {}
 
 impl cmp::PartialOrd for ValueID {
     fn partial_cmp(&self, other: &ValueID) -> Option<Ordering> {
-        let is_less_than = unsafe { extern_value_id::value_id_less_than(self.ptr, other.ptr) };
+        let is_less_than = unsafe {
+            let cvid_self = extern_value_id::value_id_from_packed_id(self.home_id, self.id);
+            let cvid_other = extern_value_id::value_id_from_packed_id(other.home_id, other.id);
+            extern_value_id::value_id_less_than(&cvid_self, &cvid_other)
```

Comment:
I don't understand here: you're creating a C ValueID here, and then once again in the Cpp wrapper.